src/w3d/lib/coord.h

```diff
@@ -22,104 +22,66 @@ using GameMath::Sqrt;
 class Coord2D
 {
 public:
-    Coord2D() : x(0.0f), y(0.0f) {}
-    Coord2D(float x_val, float y_val) : x(x_val), y(y_val) {}
```

Comment:
Revert this is much cleaner and more performant

Reply:
My feeling here after being bitten by a but caused by default initialisation to 0.0 being added is that leave the default constructor as uninitialised and keep the value initialising constructor so the caller can decide to intialise it to 0.0 or any other value at their discretion.

Reply:
I disagree with bringing back constructors in this class and think initialization should be explicit.